### PR TITLE
Removing GT_STORE_DYN_BLK, part 1

### PR DESCRIPTION
**Pull request by EgorBo**

### CODE REVIEW

**Comment by EgorBo** [comment]
I think we can actually merge it as is? [Diffs](https://dev.azure.com/dnceng-public/public/_build/results?buildId=574222&view=ms.vss-build-web.run-extensions-tab) are almost none.

I think it makes sense to separate function change (this) and the upcoming -1000 LOC clean up to actually remove `GT_STORE_DYN_BLK` (I want the clean up PR to have zero diff).

@dotnet/jit-contrib PTAL

File: src/coreclr/jit/importer.cpp [hunk 1]
```diff
@@ -10324,6 +10324,28 @@ void Compiler::impImportBlockCode(BasicBlock* block)
...
+#ifndef TARGET_X86
+                    const bool     isVolatile = (indirFlags & GTF_IND_VOLATILE) != 0;
+                    const unsigned helper     = opcode == CEE_INITBLK ? CORINFO_HELP_MEMSET : CORINFO_HELP_MEMCPY;
+#ifdef TARGET_64BIT
+                    op3 = gtNewCastNode(TYP_I_IMPL, op3, /* fromUnsigned */ true, TYP_I_IMPL);
+#endif
+                    if (isVolatile)
+                    {
+                        // Wrap with memory barriers: full-barrier + call + load-barrier
+                        impSpillSideEffects(true, CHECK_SPILL_ALL DEBUGARG("spilling side-effects"));
```
Comment (jakobbotsch) [inline src/coreclr/jit/importer.cpp:10338]: I think this needs to be called before we pop `op1`, `op2` and `op3` -- probably need to move the pops into each of the if/else.

Comment (EgorBo) [reply src/coreclr/jit/importer.cpp:10338]: Ah, right

Comment (EgorBo) [reply src/coreclr/jit/importer.cpp]: Addressed

**Review by jakobbotsch — APPROVED** [review]

**Pull request merged**